### ISSUE TITLE
Support dbt docs natively with Dagster

### ISSUE THREAD

**Issue by rexledesma** [dagster-io/dagster#15913]
### What's the use case?

How do we accommodate dbt users who are using their project's static dbt docs site?

### Ideas of implementation

Option 1:
- Upload a static site on Dagster Cloud, taking advantage of [Supercharging dbt docs](https://github.com/dagster-io/supercharged-dbt-docs)
- Generate links to it from the asset metadata

Option 2:
- Support dbt docs + metadata natively in the Dagster asset graph

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by OneCyrus** [comment]
end to end metadata in the dagster UI would be my dream. especially if it could show the table schema which is actually materialized in each dbt model instead of just the manually conifgured schema model.

**Comment by rexledesma** [comment]
Here, we went with option 2:
- We created a new overview page for an asset in the Asset Catalog: https://github.com/dagster-io/dagster/pull/19739
- We now show the column schema for a materialized dbt model, regardless if the user has defined it up front in their dbt `schema.yml` files: https://github.com/dagster-io/dagster/pull/19631. See [our documentation](https://docs.dagster.io/integrations/dbt/reference#emit-column-schema-as-materialization-metadata-) for setting this up with `dagster-dbt`.
- We'll soon show the column schema alongside any user-defined descriptions from the `schema.yml`, if present: https://github.com/dagster-io/dagster/pull/20491.
- We'll soon add support for other metadata, like owners, dbt table relation names, and more!
